cpp/include/cudf/aggregation.hpp

```diff
@@ -230,10 +230,13 @@ std::unique_ptr<aggregation> make_collect_list_aggregation(
  * @param null_handling Indicates whether to include/exclude nulls during collection
  * @param nulls_equal   Flag to specify whether null entries within each list should be considered
  * equal
+ * @param nans_equal    Flag to specify whether NaN values in floating point column should be
+ * considered equal
  */
 std::unique_ptr<aggregation> make_collect_set_aggregation(
   null_policy null_handling = null_policy::INCLUDE,
-  null_equality null_equal  = null_equality::EQUAL);
+  null_equality nulls_equal = null_equality::EQUAL,
+  nan_equality nans_equal   = nan_equality::UNEQUAL);
```

Comment:
It's a bit weird to me we're defaulting to nulls being equal and NaNs being unequal.

Reply:
nulls being equal is the default behavior everywhere in cudf, and nans being unequal is the (IEE754) standard.

Reply:
good enough for me